featurebyte/feast/utils/materialize_helper.py

```diff
@@ -81,13 +86,25 @@ def silent_tqdm_builder(length: int) -> tqdm:
         feature_view.projection.features, columns
     )
 
+    # FIXME: This patch modifies the cache used by the SnowflakeOfflineStore to introduce
+    # ttl of 10 minutes to recover from a stale connection.
+    if feature_store.config.offline_store.type == "snowflake.offline":
+        key = str(feature_store.config.offline_store)
+        if key not in feast_snowflake_session_cache:
+            feast_snowflake_session_cache[key] = {}
+        snowflake_session_cache = feast_snowflake_session_cache[key]
+    else:
+        snowflake_session_cache = {}
+
     # FIXME: This patch is related to an implementation detail of RedisOnlineStore: it checks
     # whether the feature table's stored feature timestamp is the same as current feature timestamp,
     # and if so skip the update. This doesn't work for partial materialization because a feature
     # table's stored feature timestamp is always up-to-date except on the first materialization run.
     # Patching this effectively skips the check, but a better solution might be to override the
     # implementation of RedisOnlineStore.online_write_batch().
-    with patch("google.protobuf.timestamp_pb2.Timestamp.ParseFromString"):
+    with patch("google.protobuf.timestamp_pb2.Timestamp.ParseFromString"), patch(
+        "feast.infra.utils.snowflake.snowflake_utils._cache", snowflake_session_cache
+    ):
```

Comment:
🤔 Curious whether this patching works as `_cache` is a global object and we import `feast` in this module.

Reply:
It works in integration tests - subsequent calls retrieves cached session